deploy/role.yaml

```diff
@@ -18,6 +18,23 @@ metadata:
   name: network-operator
   namespace: REPLACE_NAMESPACE
 rules:
+  - apiGroups:
```

Comment:
can you explain why you needed to add these rules for the network operator namespace ?

all resources should be deployed in the fixed namespace:  "mlnx-network-operator-resources"

Reply:
also there is already an entry below which contains pods so the `pods` entry is duplicate

Reply:
Fixes the duplicate
Some of these permissions are need to be passed to the multus, network operator needs to own the permissions to provide them for multus and other daemonsets that it deploys

Reply:
i see, so can you confirm you are adding the minimal set of permissions ?

Reply:
This PR only adds the needed permissions for the new features.